### PR TITLE
Fetch security-agent metadata from the security-agent process

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

Fetch the `security-agent` metadata from the security agent process itself rather than the local configuration. In a k8s context, the security-agent runs in a different container with potentially a different config (the security-agent API is available from the core-agent container).

### Motivation

This fixes discrepancies in the Fleet Automation product when using the Agent in k8s.

### Describe how to test/QA your changes

Test that `feature_cspm_enabled` and `feature_cspm_host_benchmarks_enabled` correctly reflect the security-agent configuration when using our official helm chart and datadog-operator.

To see the payload: from the core-agent container `agent diagnose show-metadata inventory-agent`

For team agent-security:
- Check that the two metadata  `feature_cspm_enabled` and `feature_cspm_host_benchmarks_enabled` are correctly sent when deploying the security-agent with helm chart. 

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by paulcacheux — APPROVED** [review]
Thanks a lot !

**Review by 0intro — APPROVED** [review]
It looks good to me. Thanks.

File: comp/metadata/inventoryagent/inventoryagentimpl/inventoryagent_test.go [hunk 1]
```diff
@@ -251,3 +252,31 @@ func TestConfigRefresh(t *testing.T) {
 	pkgconfig.Datadog.Set("inventories_max_interval", 10*time.Minute, pkgconfigmodel.SourceAgentRuntime)
 	assert.True(t, ia.RefreshTriggered())
 }
+
+func TestFetchSecurityAgentError(t *testing.T) {
```
Comment (sgnn7) [inline comp/metadata/inventoryagent/inventoryagentimpl/inventoryagent_test.go:256]: @hush-hush Drive-by comment: can we add a test for the error condition too?

File: pkg/config/fetcher/from_processes.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+// FetchSecurityAgentConfig fetch the configuration from the security-agent process by querying its HTTPS API
+func FetchSecurityAgentConfig(config config.Reader) (string, error) {
+	err := util.SetAuthToken()
+	if err != nil {
+		return "", err
+	}
+
+	c := util.GetClient(false)
+
+	apiConfigURL := fmt.Sprintf("https://localhost:%v/agent/config", config.GetInt("security_agent.cmd_port"))
```
Comment (pgimalac) [inline pkg/config/fetcher/from_processes.go:26]: The core-agent needs to have the proper value for `security_agent.cmd_port`, are we sure it's the case in containerized setup ?

**Review by pgimalac — APPROVED** [review]
LGTM

File: pkg/config/fetcher/from_processes.go (hunk 2 above)
Comment (hush-hush) [reply pkg/config/fetcher/from_processes.go:26]: Yes, we already correctly setup IPC with our helm chart (ie mounting sockets and opendin ports).

File: comp/metadata/inventoryagent/inventoryagentimpl/inventoryagent.go [hunk 3]
```diff
@@ -238,6 +252,37 @@ func (ia *inventoryagent) initData() {
...
+		GetBool(string) bool
+	}
+
+	securityCfg := configGetter(ia.conf)
+	// We query the configuration from the security agent itself to have accurate data. If the security-agent isn't
+	// available we fallback on the current configuration.
+	if securityConfig, err := fetchSecurityConfig(ia.conf); err == nil {
+		cfg := viper.New()
+		cfg.SetConfigType("yaml")
+		if err = cfg.ReadConfig(strings.NewReader(securityConfig)); err != nil {
```
[the remark this answers is not in the document]
Comment (hush-hush) [reply comp/metadata/inventoryagent/inventoryagentimpl/inventoryagent.go:272]: This is a good point, the scrubber should produce a valid yaml but I'll double check with security that it's the case.

File: cmd/security-agent/subcommands/config/config.go [hunk 4]
```diff
@@ -139,19 +140,13 @@ func getSettingsClient(_ *cobra.Command, _ []string) (settings.Client, error) {
...
 
-func showRuntimeConfiguration(_ log.Component, _ config.Component, _ secrets.Component, params *cliParams) error {
-	c, err := params.getClient(params.command, params.args)
-	if err != nil {
-		return err
-	}
-
-	runtimeConfig, err := c.FullConfig()
+func showRuntimeConfiguration(_ log.Component, _ config.Component, _ secrets.Component, _ *cliParams) error {
+	runtimeConfig, err := fetcher.FetchSecurityAgentConfig(pkgconfig.Datadog)
```
Comment (GustavoCaso) [inline cmd/security-agent/subcommands/config/config.go:144]: Could we use the `config.Component` rather the `pkgconfig`? 

Mostly to make sure we standardize the use of the config component

File: comp/metadata/inventoryagent/inventoryagentimpl/inventoryagent.go [hunk 5]
```diff
@@ -238,6 +252,37 @@ func (ia *inventoryagent) initData() {
 	ia.data["system_probe_protocol_classification_enabled"] = getBoolSysProbe("network_config.enable_protocol_classification")
 	ia.data["system_probe_gateway_lookup_enabled"] = getBoolSysProbe("network_config.enable_gateway_lookup")
 	ia.data["system_probe_root_namespace_enabled"] = getBoolSysProbe("network_config.enable_root_netns")
+
+	ia.data["feature_dynamic_instrumentation_enabled"] = getBoolSysProbe("dynamic_instrumentation.enabled")
+
+	ia.refreshMetadata()
```
Comment (GustavoCaso) [inline comp/metadata/inventoryagent/inventoryagentimpl/inventoryagent.go:258]: Why do we add this indirection rather than calling `fetchSecurityAgentMetadata` directly?

File: pkg/config/fetcher/from_processes.go (hunk 2 above)
Comment (pgimalac) [reply pkg/config/fetcher/from_processes.go:26]: Nice 👍

File: comp/metadata/inventoryagent/inventoryagentimpl/inventoryagent.go (hunk 5 above)
Comment (hush-hush) [reply comp/metadata/inventoryagent/inventoryagentimpl/inventoryagent.go:258]: We're going to do the same for all other processes. `refreshMetadata` is going to be expended very soon

**Review by GustavoCaso — APPROVED** [review]
